source/mod_ffield.f90

```diff
@@ -0,0 +1,937 @@
+  logical         , allocatable, private, save :: ffInQuad(:)  ! (1:npart)        Check if particle enter the Quad
+  integer         , allocatable, public, save :: ffindex(:)    ! (0:nele)         Table with the index of the Quad in our study (0 = not studied)
+  integer         , allocatable, public, save :: ffQ2File(:,:) ! (1:ffNLn, 1:2)   Link Quad/Files
+!  real(kind=fPrec), allocatable, public, save :: ffParam(:,:)  ! (1:ffNLFile,1:6) Kin, Lin, Corin, Kex, Lex, Corex
+  real(kind=fPrec), allocatable, public, save :: ffParam(:,:)  ! (1:ffNLFile,1:2) Lgth. in Quadrupoles and total lgth.
+  character(len=:), allocatable, public, save :: ffQNames(:)   ! (1:ffNLn)        Name of Quadrupoles
+  character(len=:), allocatable, public, save :: ffMSNames(:)  ! (1:ffMSn)        Name of Multipoles skip
+  character(len=:), allocatable, public, save :: ffFNames(:)   ! (1:ffNLFile)     Name of Files
+
+  type(ffTable_n_Track), allocatable, public, save :: ffTable(:)  ! (1:ffNLFile)
+  ! ------------------------------------------------------------------------------------------------ !
+  
+  contains
+
+
+  ! ================================================================================================ !
+  !  Init the module
+  !  B. Dalena, T. Pugnat and A. Simona from CEA
+  !  V.K. Berglyd Olsen, BE-ABP-HSS
+  !  Last modified: 2018-10-18
+  ! ================================================================================================ !
+  subroutine ffield_mod_init(npart, nele)
+    ! Mod from SixTrack
+    ! ---------------------------------------------------------------------------------------------- !
+    use mod_alloc,           only : alloc
+    use numerical_constants, only : zero
+    use parpro,              only : mNameLen, str_nmSpace, mStrLen, str_dSpace
+
+    implicit none
+
+    ! interface variables
+    ! ---------------------------------------------------------------------------------------------- !
+    integer, intent(in) :: npart, nele
+    
+    
+    ! initialization of the variable
+    ! ---------------------------------------------------------------------------------------------- !
+    ff_status   =0
+    ffNLn       =1
+    ffNLn_max   =20
+    ffMSn       =1
+    ffMSn_max   =20
+    ffNLFile    =1
+    ffNLFile_max=20
+
+    
+    ! allocation of the memory
+    ! ---------------------------------------------------------------------------------------------- !
+    call alloc(ffInQuad, npart,           .false., 'ffInQuad')
+    call alloc(ffindex,  nele,            0,       'ffindex')
+    call alloc(ffQ2File, ffNLFile_max, 2, 0,       'ffQ2File')
+    call alloc(ffParam,  ffNLn_max,    2, zero,    'ffParam')
+
+    call alloc(ffQNames,  mNameLen, ffNLn_max,    str_nmSpace, 'ffQNames')
+    call alloc(ffMSNames, mNameLen, ffMSn_max,    str_nmSpace, 'ffMSNames')
+    call alloc(ffFNames,  mStrLen,  ffNLFile_max, str_dSpace,  'ffFNames')
+
```

Comment:
Done but ffFNames is now of size FFFNameLgth (equal to 300) in order to save the full path to the vector potential's files.

Reply:
In that case this should be set to 255, which is the maximum for Windows and therefore also BOINC.

Reply:
In that case this should be set to 255, which is the maximum for Windows and therefore also BOINC.

Reply:
I modified parpro such that mFNameLen=255 when the flag FFIELD is activated. This shouldn't cause any problem for BOINC from what I understand.

Reply:
No, definitely don't do that. Global variables should not be modified by additional modules. We generally don't want features behind compiler flags in SixTrack, and we're working to remove those who are. Having numerous compiler options makes it increasingly difficult to maintain the code. We will look into removing the FFIELD flag, and this would then be a hindrance.

Reply:
If you need to be able to describe a full path. then I suggest you make your own parameter for this inside the FFIELD space. You can set that to 255. The mFNameLen parameter is intended for actual file names, not full file paths. I understand you need full file paths.